### PR TITLE
feat: Add function removePoliciesWithEffected

### PR DESCRIPTION
**Pull request by Edmond-J-A**
Signed-off-by: Edmond <edomondja@gmail.com>

### CODE REVIEW

**Review by rushitote — APPROVED** [review]
lgtm

File: src/model/Policy.lua [hunk 1]
```diff
@@ -304,17 +304,26 @@ function Policy:removePolicies(sec, ptype, rules)
...
+    for _,rule in pairs(rules) do
+        for k, v in pairs(self.model[sec][ptype].policy) do
+            if Util.arrayEquals(rule, v) then
+                table.insert(effected,rule)
+                table.remove(self.model[sec][ptype].policy, k)
+                break
+            end
+        end
+    end
+    return effected
```
Comment (techoner) [inline src/model/Policy.lua:326]: @Edmond-J-A Change removePolicies() to call removePoliciesWithEffected() .

**Review by techoner — CHANGES_REQUESTED** [review]

File: src/model/Policy.lua (hunk 1 above)
Comment (Edmond-J-A) [reply src/model/Policy.lua:319]: removePolicies changed

File: src/model/Policy.lua [hunk 2]
```diff
@@ -286,35 +286,30 @@ end
...
-            end
-        end
-    end
-
-    if size > #self.model[sec][ptype].policy then
-        return true
-    else
-        return false
-    end
+    return self:removePoliciesWithEffected(sec, ptype, rules)~=0
```
Comment (rushitote) [inline src/model/Policy.lua:289]: `return #self:removePoliciesWithEffected(sec, ptype, rules)~=0`

**Review by rushitote — CHANGES_REQUESTED** [review]

**Pull request merged**